packages/select/src/presets/useLazyLoading/hook.tsx

```
@@ -0,0 +1,272 @@
+                    opened: action.payload,
+                };
+            }
+            case 'SET_QUERY_STRING': {
+                return {
+                    // Изменение queryString подразумевает сброс текущих опций.
+                    ...lazyLoadingInitialState,
+                    opened: state.opened,
+                    loading: true,
+                    queryString: action.payload,
+                };
+            }
+        }
+
+        return state;
+    };
+
+    const [
+        { opened, offset, options, loading, allOptionsLoaded, queryString },
+        dispatch,
+    ] = useReducer(lazyLoadingReducer, lazyLoadingInitialState);
+
+    const renderOption = useCallback(
+        props => (
+            <Option {...props} Checkmark={null} highlighted={loading ? false : props.highlighted} />
+        ),
+        [loading],
+    );
+
+    const skeletonOptions: OptionShape[] = useMemo(() => {
+        return Array(loading ? limit : 0)
+            .fill(0)
+            .map((_, key) => ({
+                key: `loading-${key}`,
+                disabled: true,
+                content: skeleton,
+            }));
+    }, [loading, limit, skeleton]);
+
+    const abortFetchingOptionsRef = useRef<() => void>();
+
+    const fetchNextOffsetOptions = useCallback(() => {
+        dispatch(actions.fetchOptionsStart());
+
+        new Promise<OptionsFetcherResponse>((resolve, reject) => {
+            // eslint-disable-next-line no-unused-expressions
+            abortFetchingOptionsRef.current?.();
+            abortFetchingOptionsRef.current = reject;
+            optionsFetcher(offset, limit, queryString).then(res => {
+                resolve(res);
+            });
+        })
+            .then(res => {
+                dispatch(actions.fetchOptionsSuccess(res));
+                abortFetchingOptionsRef.current = undefined;
+            })
+            .catch(
+                // eslint-disable-next-line @typescript-eslint/no-empty-function
```

nit: можно в теле функции написать /* ignore errors */

---

не понял если честно что это дает